From b5144a4ca1830ab349296ec141e877651ce08332 Mon Sep 17 00:00:00 2001
From: Julien Langou <julien.langou@ucdenver.edu>
Date: Thu, 29 Dec 2016 21:53:54 +0100
Subject: [PATCH] fixes some more complex to int conversion in the LAPACKE
 files during workspace queries See pull request #107 for an initial bunch of
 these corrections, this is a follow up. This should hopefully fix them all.

---
 LAPACKE/src/lapacke_cgelq.c  | 2 +-
 LAPACKE/src/lapacke_cgemlq.c | 2 +-
 LAPACKE/src/lapacke_cgeqr.c  | 2 +-
 LAPACKE/src/lapacke_zgelq.c  | 2 +-
 LAPACKE/src/lapacke_zgemlq.c | 2 +-
 LAPACKE/src/lapacke_zgeqr.c  | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/LAPACKE/src/lapacke_cgelq.c b/LAPACKE/src/lapacke_cgelq.c
index 5dcbe32..df1d488 100644
--- a/LAPACKE/src/lapacke_cgelq.c
+++ b/LAPACKE/src/lapacke_cgelq.c
@@ -60,7 +60,7 @@ lapack_int LAPACKE_cgelq_work( int matrix_layout, lapack_int m, lapack_int n,
     if( tsize == -1 || tsize == -2 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_C2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * lwork );
     if( work == NULL ) {
diff --git a/LAPACKE/src/lapacke_cgemlq.c b/LAPACKE/src/lapacke_cgemlq.c
index 587f2df..ae39111 100644
--- a/LAPACKE/src/lapacke_cgemlq.c
+++ b/LAPACKE/src/lapacke_cgemlq.c
@@ -65,7 +65,7 @@ lapack_int LAPACKE_cgemlq( int matrix_layout, char side, char trans,
     if( info != 0 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_C2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_float*)LAPACKE_malloc( sizeof(lapack_complex_float) * lwork );
     if( work == NULL ) {
diff --git a/LAPACKE/src/lapacke_cgeqr.c b/LAPACKE/src/lapacke_cgeqr.c
index beabeb8..264b135 100644
--- a/LAPACKE/src/lapacke_cgeqr.c
+++ b/LAPACKE/src/lapacke_cgeqr.c
@@ -60,7 +60,7 @@ lapack_int LAPACKE_cgeqr( int matrix_layout, lapack_int m, lapack_int n,
     if( tsize == -1 || tsize == -2 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_C2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_float*)
         LAPACKE_malloc( sizeof(lapack_complex_float) * lwork );
diff --git a/LAPACKE/src/lapacke_zgelq.c b/LAPACKE/src/lapacke_zgelq.c
index 327aa1d..5f12035 100644
--- a/LAPACKE/src/lapacke_zgelq.c
+++ b/LAPACKE/src/lapacke_zgelq.c
@@ -60,7 +60,7 @@ lapack_int LAPACKE_zgelq_work( int matrix_layout, lapack_int m, lapack_int n,
     if( tsize == -1 || tsize == -2 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_Z2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_double*)LAPACKE_malloc( sizeof(lapack_complex_double) * lwork );
     if( work == NULL ) {
diff --git a/LAPACKE/src/lapacke_zgemlq.c b/LAPACKE/src/lapacke_zgemlq.c
index cce7703..b47ca26 100644
--- a/LAPACKE/src/lapacke_zgemlq.c
+++ b/LAPACKE/src/lapacke_zgemlq.c
@@ -65,7 +65,7 @@ lapack_int LAPACKE_zgemlq( int matrix_layout, char side, char trans,
     if( info != 0 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_Z2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_double*)LAPACKE_malloc( sizeof(lapack_complex_double) * lwork );
     if( work == NULL ) {
diff --git a/LAPACKE/src/lapacke_zgeqr.c b/LAPACKE/src/lapacke_zgeqr.c
index 61a179d..2370f4c 100644
--- a/LAPACKE/src/lapacke_zgeqr.c
+++ b/LAPACKE/src/lapacke_zgeqr.c
@@ -60,7 +60,7 @@ lapack_int LAPACKE_zgeqr( int matrix_layout, lapack_int m, lapack_int n,
     if( tsize == -1 || tsize == -2 ) {
         goto exit_level_0;
     }
-    lwork = (lapack_int)work_query;
+    lwork = LAPACK_Z2INT( work_query );
     /* Allocate memory for work arrays */
     work = (lapack_complex_double*)
         LAPACKE_malloc( sizeof(lapack_complex_double) * lwork );
-- 
2.8.1.windows.1

